custom-update-agent/util/to_files.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package util
+
+import (
+	"github.com/eclipse-kanto/update-manager/api/types"
+	"github.com/pkg/errors"
+)
+
+// ToFiles converts a list of components into a list of files
+func ToFiles(components []*types.ComponentWithConfig) ([]*File, error) {
+	files := []*File{}
+	for _, component := range components {
+		file, err := toFile(component)
+		if err != nil {
+			return nil, errors.Wrapf(err, "invalid configuration for container %s", component.ID)
+		}
+		files = append(files, file)
+	}
+	return files, nil
+}
+
+func toFile(component *types.ComponentWithConfig) (*File, error) {
+	file := &File{
+		Name:        component.Config[0].Value,
+		DownloadURL: component.Config[1].Value,
+	}
```

Comment:
Do not rely on the order of the configurations, as the DownloadURL could be assigned with the file_name. Use the keys to assign the proper fields.
```suggestion
	file := &File{}
	for _, kvPair := range component.Config {
		if kvPair.Key == "file_name" {
			file.Name = kvPair.Value
		}
		if kvPair.Key == "download_url" {
			file.DownloadURL = kvPair.Value
		}
	}
```